nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestRenditionsObject.java

```diff
@@ -0,0 +1,225 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thomas Roger
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.common.utils.FileUtils;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.thumbnail.ThumbnailService;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus.State;
+import org.nuxeo.ecm.core.io.marshallers.json.JsonAssert;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(AutomationFeature.class)
+@Deploy("org.nuxeo.ecm.core.management")
+@Deploy("org.nuxeo.ecm.platform.tag")
+@Deploy("org.nuxeo.ecm.platform.picture.api")
+@Deploy("org.nuxeo.ecm.platform.picture.core")
+@Deploy("org.nuxeo.ecm.platform.picture.convert")
+@Deploy("org.nuxeo.ecm.platform.convert")
+@Deploy("org.nuxeo.ecm.platform.types.api")
+@Deploy("org.nuxeo.ecm.platform.types.core")
+@SuppressWarnings("unlikely-arg-type")
+public class TestRenditionsObject extends ManagementBaseTest {
+
+    @Inject
+    protected CoreSession session;
+
+    public static final String PATH = "site/management/renditions";
+
+    public static final String OK = "ok";
+
+    protected static final String ENTITY_TYPE = "entity-type";
+
+    protected static final String BULK_STATUS = "bulkStatus";
+
+    protected static final String COMMAND_ID = "commandId";
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Inject
+    protected ThumbnailService thumbnailService;
+
+    @Test
+    public void testRendtionsPostPicturesRecompute() throws IOException {
+        // test the picture views are computed correctly and trashing them
+        DocumentRef docRef = beforePictureViewsRecompute();
+
+        // generating new picture views
+        runBulkAction("/pictures/recompute");
+
+        // checking the picture views are recomputed in the session
+        afterPicturesRecompute(docRef);
```

Comment:
What do you think about putting `before*` and `after*` in the test methods as we call them once?

Reply:
They are in the test in the Operations tests `TestRecomputeThumbnails` `TestRecomputePictureViews` and I wished those were separated so I can get faster to the meaty part when segmented that way.
I leave it to you to decide.